NavigationReactNative/src/ios/NVBarTitleView.m

```diff
@@ -0,0 +1,40 @@
+#import "NVBarTitleView.h"
+
+#import <React/RCTUIManager.h>
+#import <React/UIView+React.h>
+
+@interface NVBarTitleView ()
+
+@property (nonatomic, weak) RCTBridge *bridge;
+
+@end
+
+@implementation NVBarTitleView
+
+- (instancetype)initWithBridge:(RCTBridge *)bridge {
+    if (self = [super init]) {
+        self.bridge = bridge;
+    }
+    return self;
+}
+
+- (void)setFrame:(CGRect)frame {
+    BOOL frameUpdated = !CGRectEqualToRect(self.frame, frame);
+    [super setFrame:frame];
+    
+    if (self.reactTag && frameUpdated) {
+        [self.bridge.uiManager setSize:self.bounds.size forView:self];
+    }
+}
+
+- (void)didMoveToWindow {
+    [super didMoveToWindow];
+    
+    UIViewController *reactViewController = self.reactViewController;
+    if (reactViewController && self.window) {
+        self.frame = reactViewController.navigationController.navigationBar.bounds;
```

Comment:
Why not remove the frame sizing and the `setFrame` override and leave it up to the user to set the width and height via the style prop?
```jsx
<BarTitleViewIOS style={{width:200, height: 30}}>
  <Text>Customized TitleView</Text>
</BarTitleViewIOS>
```

Reply:
Oops, this is a mistake. I thought user always want a titleView taking all the empty space.

Letting user set the style is better, I'll remove this line.

Reply:
👍 You can safely remove the `setFrame` method and you don't need the `bridge` anymore

Reply:
When the `frame` of `titleView` is not set explicitly, iOS system will stretch the `titleView` to take all the space excluding `backButton` and `barButtonItems`.

And this happen outside of React's world. So here I think we need the `bridge` to notify React to correctly layout `BarTitleViewIOS`'s children.

In my experiment, this has effects when `BarTitleViewIOS`'s style is not set.

Reply:
Can you show me how to get the problem with the children layout if we remove the `setFrame` and `bridge`, please?

Reply:
Of course, my code in `sample/zoom/Detail.js`:

~~~typescript
{ /* Truncated, same as zoom/Detail.js */ }
<ScrollView contentInsetAdjustmentBehavior="automatic">
  <NavigationBarIOS title="Color">
    <BarTitleViewIOS style={styles.titleView}>
      <Text style={{fontSize:18}}>Color</Text>
      <View style={{backgroundColor: color, width: 28, height: 28}}/>
    </BarTitleViewIOS>
    <RightBarIOS>
	   <BarButtonIOS systemItem='cancel' onPress={() => {
	     stateNavigator.navigateBack(1);
	   }} />
	 </RightBarIOS>
  </NavigationBarIOS>
	{ /* Truncated, same as zoom/Detail.js */ }
</ScrollView>

const styles = Style.create({
  titleView: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  // Truncated, same as zoom/Detail.js
})
~~~

Here I'm adding two views as children of `BarTitleViewIOS`, they are expected to be centered in their parent.

If we keep the `setFrame` and `bridge`:
![with-bridge](https://user-images.githubusercontent.com/13307997/64325103-1e61a480-cffa-11e9-9618-cf94ae11fc5e.png)


After removed `setFrame` and `bridge`, the `navigationBar` looks like this:
<img width="302" alt="without-bridge" src="https://user-images.githubusercontent.com/13307997/64325116-24578580-cffa-11e9-9a22-4fd341b84ef9.png">

I add the logging code in `setFrame` and `setBounds` method and run the sample, here's the result when navigated to `Detail`:

~~~
NVBarTitleView's frame: {{0, 0}, {0, 0}}
NVBarTitleView's bounds: {{0, 0}, {375, 28}} // React layout happens here.
NVBarTitleView's frame: {{0, 0}, {279.33333333333331, 28}} // frame set by iOS
~~~

Reply:
Thanks for such a detail reply! I'll work through your example.
By the way, I love what you're doing with the zoom example. It's a great way to demo the feature

Reply:
The funny thing is that setting position to 'absolute' on the `NavigationBarIOS` component also fixes this problem! Try removing the `setFrame` with the following style
````jsx
<NavigationBarIOS title="Color" style={{position: 'absolute'}}>
```

Reply:
Oh, that's reasonable. 
The size of `BarTitleViewIOS` is determined by it's children after it's parent's `position` set to `absolute`

I'll push the latest change. Thanks !